### PR TITLE
feat(common,developer,core): add uset section 🙀

### PR DESCRIPTION
**Pull request by srl295**
- not emitted yet, basic scaffolding
- expanded type field in elem:  str/char/set

For: #7377

@keymanapp-test-bot skip

### CODE REVIEW

**Review by mcdurdin — APPROVED** [review]
LGTM -- only nits

File: common/web/types/src/kmx/kmx-plus-builder/build-uset.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { constants } from "@keymanapp/ldml-keyboard-constants";
+import { KMXPlusData, StrsItem } from "../kmx-plus.js";
+import { build_strs_index, BUILDER_STRS } from "./build-strs.js";
+import { BUILDER_SECTION } from "./builder-section.js";
+
+
+interface BUILDER_USET_USET {
+  range: number;
+  count: number;
+  pattern: number; // str
```
[the remark this answers is not in the document]
Comment (srl295) [reply common/web/types/src/kmx/kmx-plus-builder/build-uset.ts:10]: I wondered about that, probably a good idea.

File: common/web/types/src/kmx/kmx-plus-builder/build-uset.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  const ranges : BUILDER_USET_RANGE[] = [];
+
+  const usets : BUILDER_USET_USET[] = [];
+
+  // collect all sets
+  // in pattern order
+  patterns.forEach(pattern => {
+    // find the set
+    const set = kmxplus?.uset?.usets.find(s => s.uset.pattern == pattern);
```
[the remark this answers is not in the document]
Comment (srl295) [reply common/web/types/src/kmx/kmx-plus-builder/build-uset.ts:45]: OK, just sort first.  :+1:

File: common/web/types/src/kmx/kmx-plus-builder/build-uset.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (srl295) [reply common/web/types/src/kmx/kmx-plus-builder/build-uset.ts:10]: #9079

File: core/src/ldml/C7043_ldml.md [hunk 3]
```diff
@@ -574,7 +574,40 @@ For each variable,
   -  1 : set
   -  2 : unicodeSet
 
+TODO-LDML: note that at present, unicodeSet variables are not stored using the `uset` type
+
 Items are sorted by id
 
+### C7043.2.18 `uset` Unicode Sets
+
```
[the remark this answers is not in the document]
Comment (srl295) [reply core/src/ldml/C7043_ldml.md:589]: done

File: common/web/types/src/kmx/kmx-plus-builder/build-uset.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (srl295) [reply common/web/types/src/kmx/kmx-plus-builder/build-uset.ts]: done

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.129-alpha](https://keyman.com/downloads/releases/alpha/17.0.129)

**Pull request merged**